observability/otel/otel-collector/otel-collector-connectors.adoc

```diff
@@ -8,6 +8,112 @@ toc::[]
 
 A connector connects two pipelines. It consumes data as an exporter at the end of one pipeline and emits data as a receiver at the start of another pipeline. It can consume and emit data of the same or different data type. It can generate and emit data to summarize the consumed data, or it can merely replicate or route data.
 
+[id="count-connector_{context}"]
+== Count Connector
+
+The Count Connector counts trace spans, trace span events, metrics, metric data points, and log records in exporter pipelines and using the following default metric names:
+
+* `trace.span.count`
+* `trace.span.event.count`
+* `metric.count`
+* `metric.datapoint.count`
+* `log.record.count`
+
+:FeatureName: The Count Connector
+include::snippets/technology-preview.adoc[]
+
+.OpenTelemetry Collector custom resource (CR) with an enabled Count Connector
+[source,yaml]
+----
+# ...
+  config: |
+    receivers:
+      otlp:
+        protocols:
+          grpc: # <1>
+            endpoint: 0.0.0.0:4317
+    exporters:
+      prometheus: # <2>
+        endpoint: 0.0.0.0:8889
+    connectors:
+      count:
+    service:
+      pipelines:
+        traces/in:
+          receivers: [otlp]
+          exporters: [count] # <3>
+        metrics/out:
+          receivers: [count] # <4>
+          exporters: [prometheus]
+# ...
+----
+<1> This example uses the OTLP Receiver.
+<2> This example uses the Prometheus Exporter.
+<3> In this example, the Count Connector is configured to receive spans as an exporter.
+<4> In this example, the Count Connector is configured to emit generated metrics as a receiver.
+
+The Count Connector can count `spans`, `spanevents`, `metrics`, `datapoints`, and `logs` under defined conditions:
+
+.Example OpenTelemetry Collector CR for the Count Connector to count spans by conditions
+[source,yaml]
+----
+# ...
+  config: |
+    connectors:
+      count:
+        spans: # <1>
+          cluster.prod.event.count:
+            description: "<your_custom_description>"
+            conditions:
+              - 'attributes["env"] == "dev"'
+              - 'name == "devevent"'
+# ...
+----
+<1> In this example, conditions are defined for spans.
+
+The Count Connector can count `spans`, `spanevents`, `metrics`, `datapoints`, and `logs` by defined attributes:
+
+.Example OpenTelemetry Collector CR for the Count Connector to count logs by attributes
+[source,yaml]
+----
+# ...
+  config: |
+    connectors:
+      count:
+        logs: # <1>
+          my.log.count:
+            description: "<your_custom_description>"
+            attributes:
+              - key: env
+                default_value: unknown # <2>
+# ...
+----
+<1> In this example, attributes are defined for logs.
+<2> The `default_value` field defines a default value when the attribute is not set.
+
+=== Troubleshooting
+
+==== Count metrics not being generated as expected
+
+If the Count Connector is not generating the expected metrics, consider the following:
+
+.Procedure
```

Comment:
Is this actually a procedure or just a list of things to check? Must the checks be done in this order? How do we perform these checks?

Reply:
@kowen-rh, my apologies. I haven't edited this section yet.

Reply:
@iblancasa, cc @kowen-rh, I'd like to incorporate all of this into into the connector's description for two reasons:
1. It's better that we provide this info initially than have the customer fail and then look for troubleshooting help.
2. We don't have troubleshooting sections for the rest of the OTEL components.